cluster-autoscaler/cloudprovider/gce/templates.go

```diff
@@ -110,56 +125,57 @@ func (t *GceTemplateBuilder) BuildNodeFromTemplate(mig Mig, template *gce.Instan
 	node := apiv1.Node{}
 	nodeName := fmt.Sprintf("%s-template-%d", template.Name, rand.Int63())
 
+	kubeEnvValue, err := getKubeEnvValueFromTemplateMetadata(template)
+	if err != nil {
+		return nil, fmt.Errorf("could not obtain kube-env from template metadata; %v", err)
+	}
+
 	node.ObjectMeta = metav1.ObjectMeta{
 		Name:     nodeName,
 		SelfLink: fmt.Sprintf("/api/v1/nodes/%s", nodeName),
 		Labels:   map[string]string{},
 	}
 
-	capacity, err := t.BuildCapacity(cpu, mem, template.Properties.GuestAccelerators)
+	os := extractOperatingSystemFromKubeEnv(kubeEnvValue)
+	if os == OperatingSystemUnknown {
+		os = DefaultOperatingSystem
```

Comment:
nit: why define DefaultOperatingSystem near extractOperatingSystemFromKubeEnv, but only use it here?

Reply:
Idk. I guess I defined it here just randomly, by the type definition. 
On the second though I will change the `extractOperatingSystemFromKubeEnv` to return DefaultOperatingSystem if no is defined in `AUTOSCALER_ENV_VARS` and only return `OperatingSystemUnknown` if something actually unknown is there. Then I will raise an error if `OperatingSystemUnknown` is returned